### PR TITLE
updated repo to handle adding an array of components

### PR DESCRIPTION
**Pull request by zach-freyja**
Here's the PR. After thinking about it some more I think I've come around to not including `componentType`, and leaving it how it is. Since I'll be a couple of layers above this with what I'm working on, a simple abstraction that makes the component randomizing happen after entity creation isn't too onerous. I've left it as is just so you could see what I did. If you do happen to like it feel free to bring it in, but if not let me know and I'll just revert the change. 

Thanks for adding me to the repo!

### CODE REVIEW

**Comment by Stuhl** [comment]
Hey, thanks for the PR!

I've checked it out, I'm sharing your opinion that it's not necessary at the moment. I'm sure that randomizing values is something that happens more often than once but I think leaving this to the user is better because they might randomize in a way that the API wouldn't support. Meaning they would create their own randomization function anyways.

For others who don't randomize, it would add unecessary friction in the component design process. I like the idea to have this baked in but I think it would be better, to make this a real feature via objects than just doing it in the component defintiion:

```js
 physicsComponent.randomizeState("accelerationX")
```

Here begs the question for instance, what type of randomization should occur:

```js
physicsComponent.randomizeState("accelerationX", 1, 100) // -> a range?
physicsComponent.randomizeState("accelerationX", Math.random() * 100) // -> set the value?
physicsComponent.randomizeState("accelerationX", () => {return Math.random() * 100}) // -> use the value of the returned callback?
```

Lots of different ways to achieve this as you can see. Ideally I would go with the need of the users but you will always have users with a edge case or two, who would like to have something very flexible. At that point you ask yourself if the randomization process shouldn't just be handled by the user.

Getting more opinions would help, but atm the library is very unknown. Until more people show what problems they face I would leave this feature out for now. But the feedback certainly helped to think about this problem!


As for the example.ts, it's looking good and can be left as is. If you have the time to refactor your example code with the dq slime, I would gladly add that to the docs. As you may already heard, there aren't any users using this. I would like to change that.

If we can get your example in and make it a fully fledged tutorial with releases on others platforms like medium, youtube etc. I think we can make this lib gain a small userbase which would help to show more issues regarding the usability. If I have the time at some point, I would like to redesign the website a bit (looks wonky on mobile and generally kinda meh) and update the docs.

You mentioned the delta time of the frames, I would add this as a issue because that's actually something that's important for inconsistent frames. I don't know why I haven't added it to the lib ages ago. I guess I lost myself a bit in the API design.

**Comment by zach-freyja** [comment]
Updated the pr to remove componentType. It'll be a while before I clean up the example, but I'll get that to you once I do.

File: README.md
Comment (Stuhl) [inline README.md:1]: Line 56-58 needs to be changed

**Review by Stuhl — CHANGES_REQUESTED** [review]
Lines 56-58 need to be changed (readme.md), else LGTM

**Comment by zach-freyja** [comment]
oof, sorry I missed this. updated

**Review by Stuhl — APPROVED** [review]
LGTM

**Pull request merged**